mongo/options/clientoptions.go

```diff
@@ -360,6 +361,36 @@ func (c *ClientOptions) validate() error {
 		return fmt.Errorf("invalid server monitoring mode: %q", *mode)
 	}
 
+	// OIDC Validation
+	if c.Auth != nil && c.Auth.AuthMechanism == auth.MongoDBOIDC {
+		if c.Auth.Password != "" {
+			return fmt.Errorf("password must not be set for the %s auth mechanism", auth.MongoDBOIDC)
+		}
+		if c.Auth.OIDCMachineCallback != nil && c.Auth.OIDCHumanCallback != nil {
+			return fmt.Errorf("cannot set both OIDCMachineCallback and OIDCHumanCallback, only one may be specified")
+		}
+		if env, ok := c.Auth.AuthMechanismProperties[auth.EnvironmentProp]; ok {
+			switch env {
+			case auth.GCPEnvironmentValue:
+				fallthrough
+			case auth.AzureEnvironmentValue:
```

Comment:
Optional: Consider combining these cases.

```suggestion
			case auth.GCPEnvironmentValue, auth.AzureEnvironmentValue:
```

Reply:
I forgot this was possible :joy: thanks!